src/python/pants/backend/docker/util_rules/docker_binary_test.py

```diff
@@ -153,3 +160,55 @@ def mock_get_binary_shims(request: BinaryShimsRequest) -> BinaryShims:
     else:
         assert result.path == "/bin/docker"
         assert not result.is_podman
+
+
+def test_get_docker_with_tools(rule_runner: RuleRunner) -> None:
+    def mock_get_binary_path(request: BinaryPathRequest) -> BinaryPaths:
+        if request.binary_name == "docker":
+            return BinaryPaths("docker", paths=[BinaryPath("/bin/docker")])
+        elif request.binary_name == "real-tool":
+            return BinaryPaths("real-tool", paths=[BinaryPath("/bin/a-real-tool")])
+        else:
+            return BinaryPaths(request.binary_name, ())
+
+    def mock_get_binary_shims(request: BinaryShimsRequest) -> BinaryShims:
+        return BinaryShims(EMPTY_DIGEST, "cache_name")
+
+    def run(tools: list[str], optional_tools: list[str]) -> DockerBinary:
+        docker_options = create_subsystem(
+            DockerOptions,
+            experimental_enable_podman=False,
+            tools=tools,
+            optional_tools=optional_tools,
+        )
+        docker_options_env_aware = mock.MagicMock(spec=DockerOptions.EnvironmentAware)
+
+        nonlocal mock_get_binary_path
+        nonlocal mock_get_binary_shims
+
+        result = run_rule_with_mocks(
+            get_docker,
+            rule_args=[docker_options, docker_options_env_aware],
+            mock_gets=[
+                MockGet(
+                    output_type=BinaryPaths,
+                    input_types=(BinaryPathRequest,),
+                    mock=mock_get_binary_path,
+                ),
+                MockGet(
+                    output_type=BinaryShims,
+                    input_types=(BinaryShimsRequest,),
+                    mock=mock_get_binary_shims,
+                ),
+            ],
+        )
+
+        return typing.cast(DockerBinary, result)
```

Comment:
Was there an intention to use this return value for some assertions that was forgotten?

Just asking since it's being returned and not used by any of the three callers, which might indicate an oversight.

Reply:
Ah it is a left-over from me copying it from the earlier `test_get_docker_with_tools` in the file which did test values in that `DockerBinary` instance. Will just remove the return statement here.